bitcoin/src/blockdata/block.rs

```diff
@@ -378,12 +378,11 @@ mod tests {
 
         assert_eq!(block.bip34_block_height(), Ok(100_000));
 
-        // block with 9-byte bip34 push
-        const BAD_HEX: &str = "0200000035ab154183570282ce9afc0b494c9fc6a3cfea05aa8c1add2ecc56490000000038ba3d78e4500a5a7570dbe61960398add4410d278b21cd9708e6d9743f374d544fc055227f1001c29c1ea3b0101000000010000000000000000000000000000000000000000000000000000000000000000ffffffff3d09a08601112233445566000427f1001c046a510100522cfabe6d6d0000000000000000000068692066726f6d20706f6f6c7365727665726aac1eeeed88ffffffff0100f2052a010000001976a914912e2b234f941f30b18afbb4fa46171214bf66c888ac00000000";
+        // block with 3-byte bip34 push for height 1 (non-minimal)
```

Comment:
It's kinda hard to tell why this patch fixed the small height issue, since it was a bad block before and it's still a bad block after this patch.